src/index.js

```diff
@@ -0,0 +1,945 @@
+    row.appendChild(myKey);
+  }
+  return row;
+};
+
+// creating keyboard
+const createKeyboard = rowsArr => {
+  for (let i = 0; i < rowsArr.length; i += 1) {
+    document.querySelector('.keyboard').appendChild(createRow(rowsArr[i]));
+  }
+};
+
+createKeyboard(rows);
+
+// ------------ KEYBOARD EVENTS ------------- //
+const keyboard = document.querySelector('.keyboard');
+const text = document.querySelector('#text');
+let shiftState = 'up';
+let lang = 'eng';
+
+function backspace(str, start, end) {
+  let result = '';
+  if (start === end) {
+    result = str.slice(0, start-1) + str.slice(start+1, str.length);
+  } else {
+    result = str.replace(str.slice(start, end), '');
+  }
+  return result;
+}
+
+function myDelete(str, start, end) {
+  let result = '';
+
+  if (start === end) {
+    if (str.length === start) {
+      return str;
+    }
+    result = str.slice(0, start) + str.slice(start+1, str.length);
+  } else {
+    result = str.replace(str.slice(start, end), '');
+  }
+  return result;
+}
+
+function caseToggle() {
+  const keys = keyboard.querySelectorAll(`.${lang}`);
+  for (let i = 0; i < keys.length; i += 1) {
+    keys[i].querySelector('.up').classList.toggle('hide');
+    keys[i].querySelector('.down').classList.toggle('hide');
+  }
+
+  if (shiftState === 'up') {
+    shiftState = 'down';
+  } else {
+    shiftState = 'up';
+  }
+}
+
+function langToggle(currentCase) {
+  let keys = keyboard.querySelectorAll(`.${lang}`);
+  for (let i = 0; i < keys.length; i += 1) {
+    keys[i].classList.toggle('hide');
+    keys[i].querySelector(`.${currentCase}`).classList.toggle('hide');
+  }
+  if (lang === 'eng') {
+    lang = 'ru';
+    localStorage.setItem('lang', 'ru');
+  } else {
+    lang = 'eng';
+    localStorage.setItem('lang', 'eng');
+  }
+  keys = keyboard.querySelectorAll(`.${lang}`);
+  for (let i = 0; i < keys.length; i += 1) {
+    keys[i].classList.toggle('hide');
+    keys[i].querySelector(`.${currentCase}`).classList.toggle('hide');
+  }
+}
+
+if (localStorage.lang === 'ru') {
+  langToggle(shiftState);
+}
+
+function keyHandler(key, keyCode, keyValue) {
+  switch (keyCode) {
+    case 'ArrowUp':
+      text.value += '▲';
+      break;
+    case 'ArrowDown':
+      text.value += '▼';
+      break;
+    case 'ArrowLeft':
+      text.value += '◄';
+      break;
+    case 'ArrowRight':
+      text.value += '►';
+      break;
+    case 'Enter':
+      text.value += '\n';
+      break;
+    case 'Tab':
+      text.value += '    ';
+      break;
+    case 'Backspace':
+      text.value = backspace(text.value, text.selectionStart, text.selectionEnd);
+      break;
+    case 'Delete':
+      text.value = myDelete(text.value, text.selectionStart, text.selectionEnd);
+      break;
+    case 'CapsLock':
+      if (shiftState === 'up') {
+        key.classList.add('super-highlight');
+        caseToggle();
+      } else {
+        key.classList.remove('super-highlight');
+        caseToggle();
+      }
```

Comment:
It's bettre to move it in sepate function as was with 'delete' and `backspace` buttons

Reply:
fixed